pennylane/labs/resource_estimation/templates/subroutines.py

```diff
@@ -741,3 +923,301 @@ def resource_rep(cls, cmpr_ops, num_ctrl_wires) -> CompressedResourceOp:
+            num_control_wires (int): the number of control wires where in the indexes are specified
+            num_work_wires (int): the number of auxiliary wires used for QROM computation
+            size_bitstring (int): the length of each bitstring
+            clean (bool): if True, the work wires are not altered by the QROM operator
+
+        Returns:
+            CompressedResourceOp: the operator in a compressed representation
+        """
+        params = {
+            "num_bitstrings": num_bitstrings,
+            "num_bit_flips": num_bit_flips,
+            "num_control_wires": num_control_wires,
+            "num_work_wires": num_work_wires,
+            "size_bitstring": size_bitstring,
+            "clean": clean,
+        }
+        return CompressedResourceOp(cls, params)
+
+
+class ResourceAmplitudeAmplification(qml.AmplitudeAmplification, ResourceOperator):
+    r"""Resource class for the AmplitudeAmplification template.
+
+    Args:
+        U_op (Operator): the operator that prepares the state :math:`|\Psi\rangle`
+        U_params (dict): the parameters for the U operator
+        O_op (Operator): the oracle that flips the sign of the state :math:`|\phi\rangle` and does nothing to the state :math:`|\phi^{\perp}\rangle`
+        O_params (dict): the parameters for the O operator
+        iters (int): the number of iterations of the amplitude amplification subroutine
+        num_ref_wires (int): the number of wires used for the reflection
+        fixed_point (bool): whether to use the fixed-point amplitude amplification algorithm
+
+    Resources:
+        The resources are taken from the decomposition of ``qml.AmplitudeAmplification`` class.
+    """
+
+    # pylint: disable=too-many-arguments
+    @staticmethod
+    def _resource_decomp(
+        U_op,
+        U_params,
+        O_op,
+        O_params,
+        iters,
+        num_ref_wires,
+        fixed_point,
+        **kwargs,
+    ) -> Dict[CompressedResourceOp, int]:
+        r"""Returns a dictionary representing the resources of the operator. The
+        keys are the operators and the associated values are the counts.
+
+        Args:
+            U_op (Operator): the operator that prepares the state :math:`|\Psi\rangle`
+            U_params (dict): the parameters for the U operator
+            O_op (Operator): the oracle that flips the sign of the state :math:`|\phi\rangle` and does nothing to the state :math:`|\phi^{\perp}\rangle`
+            O_params (dict): the parameters for the O operator
+            iters (int): the number of iterations of the amplitude amplification subroutine
+            num_ref_wires (int): the number of wires used for the reflection
+            fixed_point (bool): whether to use the fixed-point amplitude amplification algorithm
+
+
+        Resources:
+            The resources are taken from the decomposition of ``qml.AmplitudeAmplification`` class.
```

Comment:
I believe this should link it.
```suggestion
            The resources are taken from the decomposition of :class:`qml.AmplitudeAmplification` class.
```